src/Orleans.Runtime/Networking/ConnectionListener.cs

```diff
@@ -131,23 +128,23 @@ public async Task StopAsync(CancellationToken cancellationToken)
                 }
 
                 var cycles = 0;
-                var closeTasks = new List<Task>();
                 var cancellationTask = cancellationToken.WhenCancelled();
-                while (this.ConnectionCount > 0)
+                for (var closeTasks = new List<Task>(); ; closeTasks.Clear())
```

Comment:
I think this construct is not as clear as the boring alternative

Reply:
Fair enough. Both look a bit awkward to me, but I think I used this version here as I introduced a similar loop into ConnectionManager (where previously each iteration created a new list).

I agree that it's an uncommon for-loop pattern, but it avoids both the possibility of accidentally skipping the `Clear` call and avoids the call for the first iteration (which in this case really doesn't matter).